### PR TITLE
Core NATS, Async Sidecar, NATS Auth

### PR DESCRIPTION
**Pull request by debajyoti-truefoundry**

### CODE REVIEW

File: async_processor/logger.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 logger.setLevel(logging.INFO)
 handler = logging.StreamHandler()
 handler.setLevel(logging.INFO)
-formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+formatter = logging.Formatter(
+    "%(asctime)s - %(name)s - %(levelname)s - %(process)d - %(message)s"
```
Comment (chiragjn) [inline async_processor/logger.py:8]: I have found adding `%(module)s:%(funcName)s:%(lineno)%d` very helpful when debugging things

Comment (debajyoti-truefoundry) [reply async_processor/logger.py]: Updated. Please check again.

**Review by nikp1172 — APPROVED** [review]

File: async_processor/types.py [hunk 2]
```diff
@@ -74,6 +74,11 @@ class KafkaSASLAuth(BaseModel):
     password: str
 
 
+class NATSUserPasswordAuth(BaseModel):
```
Comment (akashg3627) [inline async_processor/types.py:77]: NATSBasicAuth?

Comment (debajyoti-truefoundry) [reply async_processor/types.py:77]: They do not call it basic auth.

Basic Auth is an HTTP concept; NATS do not use HTTP.
https://docs.nats.io/running-a-nats-service/configuration/securing_nats/auth_intro/username_password#plain-text-passwords

**Pull request merged**